taxon_tools/BOT_TNRS.py

```diff
@@ -74,33 +71,43 @@ def process_taxon_resolve(taxon_frame):
     results_json = requests.post(url_tn, headers=headers, data=input_json.encode('utf-8'))
 
     results_raw = results_json.json()
+
     results = pd.DataFrame(results_raw)
 
     # Replacing empty strings with NaN
-    results['Overall_score'] = results['Overall_score'].replace('', pd.NA)
+    results['Overall_score'] = results['Overall_score'].replace('', 0)
 
     # Converting to float, handling non-convertible values by setting errors='coerce'
     results['match.score'] = pd.to_numeric(results['Overall_score'], errors='coerce').round(2).astype(str)
 
     # Selecting specific columns
     results = results[['Name_submitted', 'Overall_score', 'Name_matched', 'Taxonomic_status',
```

Comment:
now uses canoncial author instead of accepted author to prevent TNRS from mismatching new author with old taxonomic name.